### ISSUE TITLE
KeyPress events are not delievered to plugins

### ISSUE THREAD

**Issue by joseriosneto** [QupZilla/qupzilla#2369]
Hi,

This is my envinroment:

Arch Linux
qt5-base 5.9.0-2
qt5-webengine 5.9.0-1
QupZilla: d6b142bba
Plugins: [VimPlugin](https://github.com/joseriosneto/qupzilla-vimplugin)

I am not sure if this is the right place to file this issue. I've been developing a plugin for QupZilla and noticed the plugin stop receiving keyPress events (it does receive keyRelease though). A quick debug led me to `WebView::eventFilter` not forwarding this type of event to plugins because the target widget (ie, `obj`) is `m_rwhvqt` (?) and not `parentWidget()`. 

A quick hack to make it work was making WebView let the plugins process the event when the target widget is `m_rwhvqt` too:

```
diff --git a/src/lib/webengine/webview.cpp b/src/lib/webengine/webview.cpp
index 332925c4..783b100a 100644
--- a/src/lib/webengine/webview.cpp
+++ b/src/lib/webengine/webview.cpp
@@ -1301,7 +1301,7 @@ bool WebView::eventFilter(QObject *obj, QEvent *event)
         }
     }
 
-    if (obj == parentWidget()) {
+    if (obj == parentWidget() || obj == m_rwhvqt) {
         switch (event->type()) {
         case QEvent::KeyPress:
             HANDLE_EVENT(_keyPressEvent, QKeyEvent);
```
Is anyone experiencing this issue too? I believe something changed in qtwebengine since even previous versions of QupZilla that used to work are not working anymore. I also believe every plugin relying on keyPress events has the same issue likewise.

**Comment by nowrep** [comment]
> A quick debug led me to WebView::eventFilter not forwarding this type of event to plugins because the target widget (ie, obj) is m_rwhvqt (?) and not parentWidget().

Yes, this is intended, because we should not steal events from web page. In older Qt versions (at least in 5.8) all unhandled key events were forwared to `parentWidget()` so that's why key presses are checking for `parentWidget()` instead of `m_rwhvqt`.

Did you test that this change does not steal shortcuts from web pages (eg. some sites have `Ctrl+F` shortcut and in that case the action on page should be executed instead of QupZilla's search functionality).

**Comment by joseriosneto** [comment]
> In older Qt versions (at least in 5.8) all unhandled key events were forwared to parentWidget()

I am afraid this may not be true anymore. The target widget is always `m_rwhvqt` for me at event filter for all keyPress events.

> Did you test that this change does not steal shortcuts from web pages 

I haven't. Can you point me some site to check it?

**Comment by nowrep** [comment]
https://code.qt.io/cgit/qt/qtwebengine.git/tree/src/webenginewidgets/api/qwebenginepage.cpp?h=5.9#n385

It's still there in Qt 5.9, it just doesn't work anymore for key presses (only modified keys are passed through this function). I should open qtbug then ...

**Comment by joseriosneto** [comment]
> Did you test that this change does not steal shortcuts from web pages

It is. Github itself has shortcuts for iterating files with `hjkl`. Using the hack QupZilla doesn't respect them anymore.

**Comment by nowrep** [comment]
> It is. Github itself has shortcuts for iterating files with hjkl. Using the hack QupZilla doesn't respect them anymore.

Also this would be an issue when typing a text into a textfield (if you don't explicitly test for editable focused element).

**Comment by joseriosneto** [comment]
Indeed @nowrep. Got your point. I was trying to check where qtwebengine forwards the key event to parent widget but I am lost in the code. This function seems to be called by `WebContentsDelegateQt::HandleKeyboardEvent` but I haven't found yet where this is called. I'll just stop the yak shaving and focus in the qtbug. Did you file the bug? If not, I can file it. :)

**Comment by joseriosneto** [comment]
FYI
https://bugreports.qt.io/browse/QTBUG-61621

**Comment by nowrep** [comment]
Thanks! Sorry, I forgot about this.

**Comment by nowrep** [comment]
Fixed in Qt 5.9.2
